node/resolvers/list/index.ts

```diff
@@ -0,0 +1,111 @@
+import { mapKeyValues } from '../../utils/object'
+import ResolverError from '../../errors/resolverError'
+import { map, path, nth, filter, difference } from 'ramda'
+import { validateItems, validateListItem } from './util'
+import { acronymList, acronymListProduct, fields, fieldsListProduct } from './util'
+
+const getListItemsWithProductInfo = (items, catalog) => Promise.all(
+    map(async item => {
+      const productsResponse = await catalog.productBySku([path(['skuId'], item)])
+      const product = nth(0, productsResponse)
+      return { ...item, product }
+    }, items)
+  )
+
+const getListItems = async (itemsId, dataSources) => {
+  const { catalog, document } = dataSources
+  const items = itemsId ? await Promise.all(map(itemId => {
+    return document.getDocument(acronymListProduct, itemId, fieldsListProduct)
+  }, itemsId)) : []
+  return getListItemsWithProductInfo(items, catalog)
+}
+
+const addListItem = async (item, document) => {
+  const { DocumentId } = await document.createDocument(acronymListProduct, mapKeyValues({ ...item }))
+  return DocumentId
+}
+
+const addItems = async (items = [], dataSources) => {
+  const { document } = dataSources
+  validateItems(items, dataSources)
+  const promises = map(async item => {
+    return addListItem(item, document)
+  }, items)
+  return Promise.all(promises)
+}
+
+const deleteItems = (items, document) => (
+  items && items.forEach(item => document.deleteDocument(acronymListProduct, path(['itemId'], item)))
+)
+
+const updateItems = async (items, dataSources) => {
+  const { document } = dataSources
+  const itemsToBeDeleted = filter(item => path(['itemId'], item) && path(['quantity'], item) == 0, items)
+  const otherItems = difference(items, itemsToBeDeleted)
+  const itemsUpdated = await Promise.all(await map(async item => {
+    const itemId = path(['itemId'], item)
+    validateListItem(items, item, dataSources)
+    if (itemId) {
+      await document.updateDocument(acronymListProduct, itemId, mapKeyValues(item))
+      return itemId
+    } else {
```

Comment:
You don't need this else statement you could just return out of the else

Reply:
Great!